lotus/sem_ops/sem_join.py

```diff
@@ -53,6 +55,20 @@ def sem_join(
 
     left_multimodal_data = task_instructions.df2multimodal_info(l1.to_frame(col1_label), [col1_label])
     right_multimodal_data = task_instructions.df2multimodal_info(l2.to_frame(col2_label), [col2_label])
+
+    sample_docs = task_instructions.merge_multimodal_info([left_multimodal_data[0]], right_multimodal_data)
+    estimated_tokens_per_call = model.count_tokens(
+        lotus.templates.task_instructions.filter_formatter(
+            sample_docs[0], user_instruction, examples_multimodal_data, examples_answers, cot_reasoning, strategy
+        )
+    )
+    estimated_total_calls = len(l1) * len(l2)
+    estimated_total_cost = estimated_tokens_per_call * estimated_total_calls
```

Comment:
This logic should be moved inside the `if safe_mode` branch so we don't do this computation unnecessarily.